explainer.md

```diff
@@ -156,11 +154,21 @@ partial interface PresentationConnection {
 };
 
 dictionary PresentationSource {
-  readonly USVString url;
+  required PresentationSourceType type;
```

Comment:
This should be readonly as you shouldn't be able to change the source type on an existing connection.

Reply:
It turns out that "Dictionaries must not be used as the type of an attribute" ([spec](https://webidl.spec.whatwg.org/#idl-dictionaries)), so we'll just use PresentationSource in the PresentationRequest ctor.

So I've added the fields directly to the PresentationConnection interface. We could wrap them into a separate interface but that feels like complication without much benefit right now.

Reply:
From a WebIDL point of view, I do think there is benefit in declaring the mirroring parameters once  in an interface, so those declarations don't have to be repeated.

Reply:
I agree. PresentationConnection and PresentationSource are an interface and a dictionary respectively with basically the same set of fields, to achieve:
1. Not requiring the user to call a ctor for creating a PresentationSource
2. Working around WebIDL's restriction that a dictionary cannot be an interface attribute

I'm not sure if we can get rid of the duplication while maintaining the above two properties.

Reply:
Can you check the WebIDL spec to see what our options are here?

Reply:
Thus far I'm not seeing a cleaner way to do it. [Implicit conversion](https://webidl.spec.whatwg.org/#es-dictionary) from a JS object can be done with dictionaries but not interfaces. So we'd want PresentationSource to be a dictionary so that websites would be able to pass a JS object in rather than having to call a ctor. 

However a dictionary cannot be used as an attribute nor can an interface inherit from a dictionary (or vice versa), so it seems to me that the parameters need to declared once each in a dictionary and an interface.